iothub/service/tests/ExceptionHandlingHelperTests.cs

```diff
@@ -17,29 +19,57 @@ public class ExceptionHandlingHelperTests
         private const string HttpErrorCodeName = "iothub-errorcode";
 
         [TestMethod]
-        public async Task GetExceptionCodeAsync_ContentAndHeadersMatch_ValidErrorCode()
+        public async Task GetExceptionCodeAsync_ContentAndHeadersMatch_NumericErrorCode_ValidErrorCode()
         {
             // arrange
-            var httpResponseMessage = new HttpResponseMessage(System.Net.HttpStatusCode.BadRequest);
+            var httpResponseMessage = new HttpResponseMessage(HttpStatusCode.BadRequest);
             var exceptionResult = new IoTHubExceptionResult
             {
-                Message = "{\"errorCode\":404001}"
+                // A read-world message in the response content which includes the numeric error code returned by the hub service.
+                Message = 
+                "{" +
+                    "\"errorCode\":404103," +
```

Comment:
This is the type we can define that I referred to above, so we can simply index in to `ErrorCode`.